### PR TITLE
Ignite 3386

### PR DESCRIPTION
**Pull request by ezhuravl**

### CODE REVIEW

File: modules/core/src/main/java/org/apache/ignite/internal/processors/datastructures/GridCacheLockImpl.java [hunk 1]
```diff
@@ -522,7 +525,11 @@ protected boolean compareAndSetGlobalState(final int expVal, final int newVal,
                                 GridCacheLockState val = lockView.get(key);
 
                                 if (val == null)
-                                    throw new IgniteCheckedException("Failed to find reentrant lock with given name: " + name);
+                                    if (reCreate) {
```
Comment (vladisav) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/datastructures/GridCacheLockImpl.java]: What happens if failoverSafe == false? I'm thinking, if user opted for semaphore that explicitly throws exceptions if something goes wrong, do we need to inform him that new state has been created?
Also I think we should write to log if new state is created.

File: modules/core/src/test/java/org/apache/ignite/internal/GridCacheRecreateLockTest.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
+package org.apache.ignite.internal;
```
Comment (vladisav) [inline modules/core/src/test/java/org/apache/ignite/internal/GridCacheRecreateLockTest.java]: Missing license header

File: modules/core/src/test/java/org/apache/ignite/internal/GridCacheRecreateLockTest.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+            return ignite;
+        }
+        catch (Exception e) {
+            assertTrue(false);
+        }
+
+        return null;
+    }
+}
```
Comment (vladisav) [inline modules/core/src/test/java/org/apache/ignite/internal/GridCacheRecreateLockTest.java]: Missing newline

File: modules/core/src/test/java/org/apache/ignite/internal/GridCacheWorkAfterRecreateLockTest.java [hunk 4]
```diff
@@ -0,0 +1,94 @@
+package org.apache.ignite.internal;
```
Comment (vladisav) [inline modules/core/src/test/java/org/apache/ignite/internal/GridCacheWorkAfterRecreateLockTest.java]: Missing license header

File: modules/core/src/test/java/org/apache/ignite/internal/GridCacheWorkAfterRecreateLockTest.java [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+            return ignite;
+        }
+        catch (Exception e) {
+            assertTrue(false);
+        }
+
+        return null;
+    }
+
+}
```
Comment (vladisav) [inline modules/core/src/test/java/org/apache/ignite/internal/GridCacheWorkAfterRecreateLockTest.java]: Missing newline

**Comment by dspavlov** [comment]
The ticket is resolved, closing PR. 
 
Please close PRs for your resolved tickets and other non-needed PRs. 
 
Alternatively, ask a committer 
- to use GitHub UI (committer can associate accounts here: https://gitbox.apache.org/setup/ )
- or ./scripts/apply_pull_request.sh from Apache Ignite source.
- or mention in commit message 'Fixes #(a number of PR)'. 
 
These methods will automatically close PR.
 
Feel free to reopen this PR if it is anyway actual.

**Comment by lakshmikanthGIT** [comment]
This PR is not available in ignite-core-2.16.0. Was it removed? 
I am facing the same issue though i supplied create=true in Ignite.reentrantLock() method.

rg.apache.ignite.IgniteCheckedException: Failed to find reentrant lock with given name:

**Comment by ravaelamanov** [comment]
@lakshmikanthGIT, apparently this commit was reverted here https://github.com/apache/ignite/commit/d20f4588f5b26ba7f36111329211e8ed6e4188dc by @niktikhonov

@ezhuravl, any explanation on why this PR was reverted?

**Pull request closed without merging**